### ISSUE TITLE
Panic when running pulumi import on Azure resources with --local login

### ISSUE THREAD

**Issue by OlegKarasik** [pulumi/pulumi#5876]
## Expected Behavior
The import call is completed

## Current Behavior
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x60 pc=0x186cdd7]

goroutine 27 [running]:
github.com/pulumi/pulumi/pkg/v2/resource/deploy.(*importer).registerExistingResources(0xc000e91b00, 0x2a449a0, 0xc000ec4f00, 0x3)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/resource/deploy/import.go:134 +0x37
github.com/pulumi/pulumi/pkg/v2/resource/deploy.(*importer).importResources(0xc000f0bb00, 0x2a449a0, 0xc000ec4f00, 0xc0001525a0, 0x204eab92a48)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/resource/deploy/import.go:275 +0x7a
github.com/pulumi/pulumi/pkg/v2/resource/deploy.(*deploymentExecutor).importResources(0xc00144e2e0, 0x2a44a60, 0xc00146c030, 0x204eab92a48, 0xc000ce6140, 0x7fffffff, 0x0, 0x0, 0x0, 0x0, ...)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/resource/deploy/deployment_executor.go:450 +0x173
github.com/pulumi/pulumi/pkg/v2/resource/deploy.(*deploymentExecutor).Execute(0xc00144e2e0, 0x2a44a60, 0xc00146c030, 0x204eab92a48, 0xc000ce6140, 0x7fffffff, 0x0, 0x0, 0x0, 0x0, ...)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/resource/deploy/deployment_executor.go:138 +0x1247
github.com/pulumi/pulumi/pkg/v2/resource/deploy.(*Deployment).Execute(...)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/resource/deploy/deployment.go:383
github.com/pulumi/pulumi/pkg/v2/engine.(*deployment).run.func1(0x2a4fa80, 0xc000ce6140, 0xc00016c3c0, 0x2a44a60, 0xc00146c030, 0xc00146c001, 0xc000e9dcf0, 0xc001464480)
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/engine/deployment.go:251 +0x287
created by github.com/pulumi/pulumi/pkg/v2/engine.(*deployment).run
        D:/a/pulumi/pulumi/src/github.com/pulumi/pulumi/pkg/engine/deployment.go:237 +0x314
```

## Steps to Reproduce
1. Login locally (using `pulumi login --local`)
2. Try to import any Azure resource (ex. `pulumi import azure:keyvault/keyVault:KeyVault example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.KeyVault/vaults/vault1`)

## Context (Environment)

I am using C# as pulumi language (2.15.0 cli) and I was trying to import Azure KeyVault resource.

**Comment by stack72** [comment]
Hi @OlegKarasik 

I am releasing pulumi v2.15.2 which I believe fixes your issue right now - it will hit package managers within an hour

Paul

**Comment by stack72** [comment]
Hi @OlegKarasik 

My apologies you are seeing this issue - I just opened a PR that fixes the problem and it will be made available in Pulumi v2.15.4 later today

Paul

**Comment by OlegKarasik** [comment]
Hi @stack72,

Thanks for a quick response. I am glad it will be fixed soon!

**Comment by OlegKarasik** [comment]
The issues seems to be not fixed. I have just tried to import AppServicePlan and got the same error. (CLI 2.15.4)

**Comment by stack72** [comment]
Hi @OlegKarasik 

I am not sure why this is the case - I just installed the latest version and can see the following:

```
▶ pulumi import  azure:keyvault/keyVault:KeyVault example /subscriptions/0282681f-7a9e-424b-80b2-96babd57a8a1/resourceGroups/qweisfjxfboy/providers/Microsoft.KeyVault/vaults/qweisfjxfboy2
Previewing import (dev)

View Live: https://app.pulumi.com/stack72/gh-5876/dev/previews/2f664381-0fec-4c45-ad1f-11b1f6350dd0

     Type                        Name         Plan
 +   pulumi:pulumi:Stack         gh-5876-dev  create
 =   └─ azure:keyvault:KeyVault  example      import

Resources:
    + 1 to create
    = 1 to import
    2 changes

Do you want to perform this import? yes
Importing (dev)

View Live: https://app.pulumi.com/stack72/gh-5876/dev/updates/1

     Type                        Name         Status
 +   pulumi:pulumi:Stack         gh-5876-dev  created
 =   └─ azure:keyvault:KeyVault  example      imported

Resources:
    + 1 created
    = 1 imported
    2 changes

Duration: 10s

Please copy the following code into your Pulumi application. Not doing so
will cause Pulumi to report that an update will happen on the next update command.

Please note, that the imported resources are marked as protected. To destroy them
you will need to remove the `protect` option and run `pulumi update` *before*
the destroy will take effect.

using Pulumi;
using Azure = Pulumi.Azure;

class MyStack : Stack
{
    public MyStack()
    {
        var example = new Azure.KeyVault.KeyVault("example", new Azure.KeyVault.KeyVaultArgs
        {
            EnabledForTemplateDeployment = true,
            Location = "westeurope",
            Name = "qweisfjxfboy2",
            ResourceGroupName = "qweisfjxfboy",
            SkuName = "standard",
            SoftDeleteEnabled = true,
            TenantId = "706143bc-e1d4-4593-aee2-c9dc60ab9be7",
        }, new CustomResourceOptions
        {
            Protect = true,
        });
    }

}
```

**Comment by OlegKarasik** [comment]
Hi @stack72,

As I can see from your code snippet you are logged into cloud account (`https://app.pulumi.com/stack72/gh-5876/dev/updates/1`), however, this issue is reproduced only if you are logged using `--local` flag i.e. when stacks are stored locally.

**Comment by stack72** [comment]
Got it - so this is specific to local backend! Ok, this is a different panic to that of what was fixed in v2.15.4

We can re-triage this issue now

Paul

**Comment by stack72** [comment]
ok, I found the issue and a PR is pending

```
▶ pulumi import azure:keyvault/keyVault:KeyVault example /subscriptions/0282681f-7a9e-424b-80b2-96babd57a8a1/resourceGroups/qweisfjxfboy/providers/Microsoft.KeyVault/vaults/qweisfjxfboy2
Enter your passphrase to unlock config/secrets
    (set PULUMI_CONFIG_PASSPHRASE or PULUMI_CONFIG_PASSPHRASE_FILE to remember):
Previewing import (dev):
     Type                        Name         Plan
 +   pulumi:pulumi:Stack         gh-5876-dev  create
 =   └─ azure:keyvault:KeyVault  example      import

Resources:
    + 1 to create
    = 1 to import
    2 changes

Permalink: file:///Users/stack72/.pulumi/stacks/dev.json
Do you want to perform this import? yes
Importing (dev):
     Type                        Name         Status
 +   pulumi:pulumi:Stack         gh-5876-dev  created
 =   └─ azure:keyvault:KeyVault  example      imported

Resources:
    + 1 created
    = 1 imported
    2 changes

Duration: 9s

Permalink: file:///Users/stack72/.pulumi/stacks/dev.json
Please copy the following code into your Pulumi application. Not doing so
will cause Pulumi to report that an update will happen on the next update command.

Please note, that the imported resources are marked as protected. To destroy them
you will need to remove the `protect` option and run `pulumi update` *before*
the destroy will take effect.

using Pulumi;
using Azure = Pulumi.Azure;

class MyStack : Stack
{
    public MyStack()
    {
        var example = new Azure.KeyVault.KeyVault("example", new Azure.KeyVault.KeyVaultArgs
        {
            EnabledForTemplateDeployment = true,
            Location = "westeurope",
            Name = "qweisfjxfboy2",
            ResourceGroupName = "qweisfjxfboy",
            SkuName = "standard",
            SoftDeleteEnabled = true,
            TenantId = "706143bc-e1d4-4593-aee2-c9dc60ab9be7",
        }, new CustomResourceOptions
        {
            Protect = true,
        });
    }

}
```
